### PR TITLE
Stats: Update promo card links to Jetpack mobile app

### PR DESCRIPTION
**Pull request by a8ck3n**
#### Proposed Changes

The current links and QR code (as pulled from the Figma design doc) direct us to the WordPress mobile app. I've updated the URLs and QR code to instead direct us to the Jetpack mobile app.

#### Testing Instructions

* Check out this branch locally.
* Spin up Storybook via `yarn workspace @automattic/components run storybook`.
* Check the Promo Card stories. Ensure the links go to the Jetpack mobile app page.
* Scan the QR code and confirm it goes to the same page.

Updated image showing minor changes.

JetpackMobilePromoCard

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #69230 

Depends on #70023

### CODE REVIEW

**Review by jsnmoon — APPROVED** [review]

File: packages/components/src/promo-card/index.tsx [hunk 1]
```diff
@@ -12,13 +12,17 @@ export default function PromoCard( { className, isMobile }: PromoCardProps ) {
...
 	const components = {
-		a: <a href="https://wp.com/apps" />,
+		a: <a href={ redirectLink } />,
 	};
 	// TODO: Fix mobile logic.
 	// Should this maybe come via a media query in the CSS?
 	// Also, does it make sense to push one app store over the other?
 	// Might make more sense to have a custom "app store" SVG image that goes to wp.com/app.
+
+	// TODO: Include QR code in bundle?
```
Comment (jsnmoon) [inline packages/components/src/promo-card/index.tsx:24]: Including the QR SVG directly will make this component less brittle, though I could also see why we might continue serving it from s0.wp.com for convenience's sake.

File: packages/components/src/promo-card/index.tsx [hunk 2]
```diff
@@ -12,13 +12,17 @@ export default function PromoCard( { className, isMobile }: PromoCardProps ) {
 	// Using useTranslate() with interpolation to set up the title/message.
 	// https://wpcalypso.wordpress.com/devdocs/packages/i18n-calypso/README.md
 	const translate = useTranslate();
+	const redirectLink = 'https://jetpack.com/app/';
 	const components = {
-		a: <a href="https://wp.com/apps" />,
+		a: <a href={ redirectLink } />,
 	};
 	// TODO: Fix mobile logic.
 	// Should this maybe come via a media query in the CSS?
```
Comment (jsnmoon) [inline packages/components/src/promo-card/index.tsx:20]: We could definitely use media queries to show the appropriate copy & image combination.

**Review by jsnmoon — APPROVED** [review]
Merged trunk in f30911b. These changes look good!

File: packages/components/src/promo-card/index.tsx (hunk 1 above)
Comment (a8ck3n) [reply packages/components/src/promo-card/index.tsx:24]: Yeah, I can see benefits to both approaches. I think I'd prefer to have the SVG data directly in the bundle, personally. I'll create a maintenance issue so we can revisit this.

File: packages/components/src/promo-card/index.tsx (hunk 2 above)
Comment (a8ck3n) [reply packages/components/src/promo-card/index.tsx:20]: I'll add a note about this one to the [main issue](https://github.com/Automattic/wp-calypso/issues/69230) so I remember to circle back to this.

**Pull request merged**